pages/blueprint/index.js

```diff
@@ -183,6 +193,12 @@ class BlueprintPage extends React.Component {
     event.preventDefault();
   }
 
+  handleEditHostnameValue(value) {
+    const validCharacters = /^[a-zA-Z0-9-.]{1,1000}$/.test(value);
```

Comment:
@larskarlitski I was fumbling my way through this one. Any suggestions are much appreciated. :-)

Reply:
I think this could be good enough:
```regex
^(\d|\w)(\d|\w|-){0,62}(\.(\d|\w)(\d|\w|-){0,62})*$
```

It checks that the first letter is a digit or a word character (and not a `-`), followed by up to 62 digits, word characters. Then optionally a dot with the same. You'd only need to check that the whole string is shorter than 253 characters.

Reply:
I included this change in my latest commit, but when I type in 63 characters my browser becomes unresponsive. Sometimes it becomes unresponsive after typing almost 63 characters and then a dot. Could you check this and see if you have the same issues?